### PR TITLE
build: Add `tag-build` recipe to Makefile

### PR DESCRIPTION
**Pull request by camscale**
Add a `tag-build` recipe to the Makefile to kick off a tag build on
GitHub Actions. Currently we cannot trigger the tag build in GitHub
Actions as the teleport repository is public so we are very careful with
credentials in it, and credentials are needed to run the `tag-build`
workflow in the `teleport.e` repository.

The new process for building a tag is to run `make update-tag` as usual,
but to then follow that with `make tag-build` to start the build.

Issue: https://github.com/gravitational/teleport/issues/20729

### CODE REVIEW

**Review by fheinecke — APPROVED** [review]

**Comment by zmb3** [comment]
Can we call it tag-build instead? This would match the GitHub workflow and avoid the ambiguity associated with the term "build tab" which is a distinctly different thing in the Go ecosystem.

**Comment by camscale** [comment]
We could call it tag-build, but I was following along with the process: `update-version`, `update-tag`, `build-tag`, `publish-tag`. Phony targets in the Makefile tend to be verbs or action-oriented. But I'm not fussed if you prefer it that way. I will also change `publish-tag` to `tag-publish`.

**Comment by camscale** [comment]
> Can we call it tag-build instead

done

File: Makefile [hunk 1]
```diff
@@ -1152,8 +1152,13 @@ $(VERSRC): Makefile
...
 # 4. Commit version changes to git
 # 5. Make sure it all builds (`make release` or equivalent)
+# 6. Run `make update-tag` to tag repos with $(VERSION)
+# 7. Run `make tag-build` to build the tag on GitHub Actions
+# 8. Run `make tag-publish` after `make-build` tag has completed to
+#    publish the built artifacts.
 #
-# After the above is done, run `make update-tag` and follow your build on Drone.
+# GHA tag builds: https://github.com/gravitational/teleport.e/actions/workflows/tag-build.yaml
+# TGA tag publish: https://github.com/gravitational/teleport.e/actions/workflows/tag-publish.yaml
```
Comment (zmb3) [inline Makefile:1161]: What is TGA?

**Review by zmb3 — APPROVED** [review]

**Review by strideynet — APPROVED** [review]
Just used this locally to trigger https://github.com/gravitational/teleport.e/actions/runs/8055743348 - works great!

**Pull request merged**